### ISSUE TITLE
Module not found StringUtils

### ISSUE THREAD

**Issue by XerxesAkaEnno** [typeorm/typeorm#8656]
## Issue Description

After updating to version 0.2.42 I get an error on my next js application which says "Module not Found ../typeorm/util/StringUtils"
Seems like its no exported any more.

### Expected Behavior
```
The StringUtils should be exported, for minor version.
```

### Actual Behavior

```
  StringUtils are not exported to implement as module
```

### Steps to Reproduce
```
For example load up a Next JS Project with Next-Auth which has Typeorm as dependency.  
```

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |          |
| Node.js version     | x.y.zzz  | 
| Typescript version  | x.y.zzz  | 
| TypeORM version     | x.y.zzz  | 

### Relevant Database Driver(s)

| DB Type              | Reproducible  |
| ---                  | ---           |
| `aurora-data-api`    | no            |
| `aurora-data-api-pg` | no            |
| `better-sqlite3`     | no            |
| `cockroachdb`        | no            |
| `cordova`            | no            |
| `expo`               | no            |
| `mongodb`            | no            |
| `mysql`              | no            |
| `nativescript`       | no            |
| `oracle`             | no            |
| `postgres`           | no            |
| `react-native`       | no            |
| `sap`                | no            |
| `sqlite`             | no            |
| `sqlite-abstract`    | no            |
| `sqljs`              | no            |
| `sqlserver`          | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 -  ✖️Yes, I have the time, and I know how to start.
 - ✖️ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✅ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by visarya** [comment]
facing the same issue

**Comment by XerxesAkaEnno** [comment]
As a workaround for projects or dependencies  following works:


**This workaround works on my pipeline.** 

1. Add to the package.json following: "typeorm": "^0.2.42",
The package wouldn't be downloaded again because an dependency already had it.

2. run `npm install`

3. run `npm remove typeorm`

4. run `npm install typeorm@0.2.41`

**Comment by Anu1601CS** [comment]
facing the same issue. Even i was on old version `^0.2.25`. My deployments are failing suddenly

**Comment by spalt08** [comment]
https://github.com/tonivj5/typeorm-naming-strategies/issues/16

**Comment by Anu1601CS** [comment]
I am wondering how its  affecting older versions ? `^0.2.25 `

**Comment by pleerock** [comment]
> I am wondering how its affecting older versions ? ^0.2.25

it doesn't affect old version, because of `^` it installs the latest right now on your machine.

**Comment by nallenscott** [comment]
Appears to be related to ESM:

<img width="822" alt="Screen Shot 2022-02-17 at 9 48 08 AM" src="https://user-images.githubusercontent.com/11765848/154506602-4563ed44-f586-4027-a346-8a61b4a16e6c.png">

**Comment by pleerock** [comment]
any hints on how to reproduce this issue at it's minimal (without other big dependencies like nestjs, etc)

**Comment by spalt08** [comment]
@pleerock Just use the `typeorm-naming-strategies` package (snake case strategy for example) in your connection config

**Comment by Anu1601CS** [comment]
> > I am wondering how its affecting older versions ? ^0.2.25
> 
> it doesn't affect old version, because of `^` it installs the latest right now on your machine.

Yes minor version issue.

**Comment by pleerock** [comment]
@giladgd thanks for urgent response, fix was released in `0.2.43`.

**Comment by k-patton** [comment]
I am seeing this issue again in version `0.2.44`

**Comment by giladgd** [comment]
@k-patton I tried to reproduce it but this problem didn't occur to me.
Could you please provide me with an example project where it happens to you?

**Comment by kenchoong** [comment]
I using typeorm 0.2.44 

use it with Nx. Having the same issue as above.

**Comment by nastari** [comment]
same problem.... sad

**Comment by adamskrodzki** [comment]
In hope that this will help anybody.

In my case problem was fixed by using node   v16.13.0  instead of v14.18.1
